api/src/main/java/io/opencensus/trace/Span.java

```diff
@@ -146,9 +146,23 @@ public final void addAnnotation(String description) {
    * higher level applications.
    *
    * @param networkEvent the network event to add.
+   * @deprecated Use {@link #addMessageEvent}.
    */
+  @Deprecated
   public abstract void addNetworkEvent(NetworkEvent networkEvent);
```

Comment:
addNetworkEvent should be implemented in terms of addMessageEvent, so that users can migrate Span subclasses to the new API, as I described in my previous comment.

Reply:
Are you suggesting implementing  `addNetworkEvent` and leave `addMessageEvent` abstract?
I am not sure if I am understanding your previous comment correctly :) 
>This method (addMessageEvent) should have a default implementation that calls addNetworkEvent, so that it doesn't break any existing subclasses, such as fake Span implementations for tests.

I myself do agree with you that it is better to implement `addNetworkEvent` using `addMessageEvent`.

Reply:
I meant that addNetworkEvent should call addMessageEvent, and addMessageEvent should call addNetworkEvent, so that either one can be overridden.  Users can start using the new version of OpenCensus while they override addNetworkEvent and can then switch to only overriding addMessageEvent.  That is how we have handled other deprecated overridable methods.

Reply:
Found a similar case:
https://github.com/census-instrumentation/opencensus-java/blob/a19d288a49a6635bfc3e99740fcf697234cd476e/api/src/main/java/io/opencensus/trace/Span.java#L112

Here `putAttributes` deprecated old API `addAttributes`. In API definition, the new API is by default implemented using old one. However, in real implementation, both of the APIs are overridden and old API is implemented using the new one.